src/consumer.rs

```diff
@@ -211,7 +213,10 @@ impl ConsumerInner {
                 .send(Err(error))
                 .expect("failed to send error to consumer");
         }
-        self.cancel()
+        if let Some(ref task) = self.task {
+            task.notify();
+        }
+        Ok(())
```

Comment:
Does this really need to be changed? I don't think sending an extra Ok(None) after the error is a problem, and it brings consistency wrt Ok(None) being the marker for finishing a stream

Reply:
The problem is that `cancel` method clears the consumer channel and the messages we just send there getting erased before `Stream::poll_next` has a chance to receive it.

Reply:
Also, I have a feeling, it's better to serve pending messages in the consumer channel before the error, instead of discarding them (`cancel` behaviour atm).